src/pset/map/input.rs

```diff
@@ -420,9 +421,33 @@ impl Input{
         ret
     }
 
+    /// Compute the issuance asset ids from pset. This function does not check
+    /// whether there is an issuance in this input. Returns (asset_id, token_id)
+    pub fn issuance_ids(&self) -> (AssetId, AssetId) {
+        let issue_nonce = self.issuance_blinding_nonce.unwrap_or(ZERO_TWEAK);
+        let entropy = if issue_nonce == ZERO_TWEAK {
+            // new issuance
+            let prevout = OutPoint {
+                txid: self.previous_txid,
+                vout: self.previous_output_index,
+            };
+            let contract_hash =
+                ContractHash::from_inner(self.issuance_asset_entropy.unwrap_or_default());
+            AssetId::generate_asset_entropy(prevout, contract_hash)
+        } else {
+            // re-issuance
+            sha256::Midstate::from_inner(self.issuance_asset_entropy.unwrap_or([0u8; 32]))
```

Comment:
In 8b41c1327d8af43f2196ce97d230c5a0719d74c3:

Should use `unwrap_or_default` here like above (or change both to `unwrap_or([8; 32])`; having two ways to write the same thing makes it look like these  do different things, which is confusing.

Reply:
Still present in 04ad35f

Reply:
Addressed in fa753b4